### PR TITLE
#17290 naming re-alignment

### PR DESCRIPTION
**Pull request by mohamedshabanofficial**
@kristapratico

### CODE REVIEW

File: sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py [hunk 1]
```diff
@@ -105,15 +105,15 @@ def _to_generated_list(targets):
...
     # pylint: disable=C0301
     """Definition for the input batch translation request.
 
     :param source_url: Required. Location of the folder / container or single file with your
      documents.
     :type source_url: str
     :param targets: Required. Location of the destination for the output.
-    :type targets: list[StorageTarget]
+    :type targets: list[TranslationTarget]
     :keyword str source_language: Language code
```
Comment (kristapratico) [inline sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py:117]: I suppose we should also rename `source_language` to `source_language_code` to be consistent.

File: sdk/documenttranslation/azure-ai-documenttranslation/samples/async_samples/sample_batch_translation_async.py [hunk 2]
```diff
@@ -94,13 +94,13 @@ async def check_documents(self, client, job_id):
...
                 print("Document at {} failed to be translated to {} language".format(
-                    document.url, document.translate_to
+                    document.translated_document_url, document.translate_to
                 ))
                 print("Document ID: {}, Error Code: {}, Message: {}".format(
                     document.id, document.error.code, document.error.message
                 ))
-                if document.url not in docs_to_retry:
-                    docs_to_retry.append(document.url)
+                if document.translated_document_url not in docs_to_retry:
```
Comment (kristapratico) [inline sdk/documenttranslation/azure-ai-documenttranslation/samples/async_samples/sample_batch_translation_async.py:102]: I think we want to save the `source_url` for the document that failed (so we can retry on it), the `translated_document_url` shouldn't exist in this scenario.

Comment (mohamedshabanofficial) [reply sdk/documenttranslation/azure-ai-documenttranslation/samples/async_samples/sample_batch_translation_async.py:102]: yeah, i mentioned it here
https://github.com/Azure/azure-sdk-for-python/issues/17288#issuecomment-804441848

File: sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py (hunk 1 above)
Comment (mohamedshabanofficial) [reply sdk/documenttranslation/azure-ai-documenttranslation/azure/ai/documenttranslation/_models.py]: done

**Review by kristapratico — APPROVED** [review]

**Pull request merged**